fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/api/IConnectionProvider.java

```diff
@@ -57,4 +56,11 @@
      * @param key
      */
     public void describe(String prefix, StringBuilder cfg, String key);
-}
+
+    /**
+     * used to control threading size.
+     */
+    public default int getPoolSize() {
+        return -1;
```

Comment:
does -1 mean no limit?

Reply:
it means there is now pool and to signal the downstream code to change.